### ISSUE TITLE
TypeError: Cannot read property 'name' of null

### ISSUE THREAD

**Issue by MrOrz** [cofacts/rumors-site#196]
View details in Rollbar: [https://rollbar.com/mrorz/rumors-site/items/10/](https://rollbar.com/mrorz/rumors-site/items/10/)

![image](https://user-images.githubusercontent.com/108608/71642107-614e3300-2ce1-11ea-9e7e-9972e94a211e.png)

```

TypeError: Cannot read property 'name' of null
  File "/srv/www/.next/server/static/z2wDCzrGvJHkWWjC83-vD/pages/reply/[id].js", line 1536, in UsedArticleItem
        editorName: user.name,
  File "/srv/www/node_modules/react-dom/cjs/react-dom-server.node.production.min.js", line 33, in c
    function Sa(a,b,d){function c(c,f){var e=ma(f,b,d),g=[],h=!1,l={isMounted:function(){return!1},enqueueForceUpdate:function(){if(null===g)return null},enqueueReplaceState:function(a,b){h=!0;g=[b]},enqueueSetState:function(a,b){if(null===g)return null;g.push(b)}},k=void 0;if(f.prototype&&f.prototype.isReactComponent){if(k=new f(c.props,e,l),"function"===typeof f.getDerivedStateFromProps){var t=f.getDerivedStateFromProps.call(null,c.props,k.state);null!=t&&(k.state=p({},k.state,t))}}else if(N={},k=f(c.props,
  File "/srv/www/node_modules/react-dom/cjs/react-dom-server.node.production.min.js", line 36, in Sa
    c(f,e)}return{child:a,context:b}}
  File "/srv/www/node_modules/react-dom/cjs/react-dom-server.node.production.min.js", line 41, in a.render
    D}else{var l=g.children[g.childIndex++],k="";try{k+=this.render(l,g.context,g.domNamespace)}catch(t){throw t;}finally{}f.length<=this.suspenseDepth&&f.push("");f[this.suspenseDepth]+=k}}return f[0]}finally{Ja.current=c,X=b}};a.prototype.render=function(a,d,c){if("string"===typeof a||"number"===typeof a){c=""+a;if(""===c)return"";if(this.makeStaticMarkup)return M(c);if(this.previousWasTextNode)return"\x3c!-- --\x3e"+M(c);this.previousWasTextNode=!0;return M(c)}d=Sa(a,d,this.threadID);a=d.child;d=d.context;
  File "/srv/www/node_modules/react-dom/cjs/react-dom-server.node.production.min.js", line 41, in a.read
    D}else{var l=g.children[g.childIndex++],k="";try{k+=this.render(l,g.context,g.domNamespace)}catch(t){throw t;}finally{}f.length<=this.suspenseDepth&&f.push("");f[this.suspenseDepth]+=k}}return f[0]}finally{Ja.current=c,X=b}};a.prototype.render=function(a,d,c){if("string"===typeof a||"number"===typeof a){c=""+a;if(""===c)return"";if(this.makeStaticMarkup)return M(c);if(this.previousWasTextNode)return"\x3c!-- --\x3e"+M(c);this.previousWasTextNode=!0;return M(c)}d=Sa(a,d,this.threadID);a=d.child;d=d.context;
  File "/srv/www/node_modules/react-dom/cjs/react-dom-server.node.production.min.js", line 53, in renderToString
    return b}(aa.Readable),Wa={renderToString:function(a){a=new Ta(a,!1);try{return a.read(Infinity)}finally{a.destroy()}},renderToStaticMarkup:function(a){a=new Ta(a,!0);try{return a.read(Infinity)}finally{a.destroy()}},renderToNodeStream:function(a){return new Va(a,!1)},renderToStaticNodeStream:function(a){return new Va(a,!0)},version:"16.8.6"},Xa={default:Wa},Ya=Xa&&Wa||Xa;module.exports=Ya.default||Ya;
  File "/srv/www/node_modules/next/dist/next-server/server/render.js", line 81, in render
            html = renderElementToString(element);
  File "/srv/www/node_modules/next/dist/next-server/server/render.js", line 296, in renderPage
                return render(renderElementToString, react_1.default.createElement(AppContainer, null,
  File "/srv/www/.next/server/static/z2wDCzrGvJHkWWjC83-vD/pages/_document.js", line 446, in ctx.renderPage
          const rendered = originalRenderPage({
  File "/srv/www/.next/server/static/z2wDCzrGvJHkWWjC83-vD/pages/_document.js", line 776, in <unknown>
          } = yield renderPage();
```

**Comment by MrOrz** [comment]
Found a way to reproduce:

Visit https://cofacts.g0v.tw/reply/5605652568811-answer

This reply triggers error because
- Used in multiple articles ([article1](https://cofacts.g0v.tw/article/5605652568811-rumor), [article2](https://cofacts.g0v.tw/artcile/5601640685411-rumor)) thus `UsedArticleItem` is rendered
- `userId` / `appId` is not correct, thus `user` field returned from GraphQL is null

**Comment by MrOrz** [comment]
Fixed by #215 , kudos to @normanlinnet
